mixnode/src/commands/init.rs

```diff
@@ -5,70 +5,64 @@ use crate::commands::*;
-            Arg::with_name(VERLOC_PORT_ARG_NAME)
-                .long(VERLOC_PORT_ARG_NAME)
-                .help("The port on which the mixnode will be listening for verloc packets")
-                .takes_value(true),
-        )
-        .arg(
-            Arg::with_name(HTTP_API_PORT_ARG_NAME)
-                .long(HTTP_API_PORT_ARG_NAME)
-                .help("The port on which the mixnode will be listening for http requests")
-                .takes_value(true),
-        )
-        .arg(
-            Arg::with_name(ANNOUNCE_HOST_ARG_NAME)
-                .long(ANNOUNCE_HOST_ARG_NAME)
-                .help("The custom host that will be reported to the directory server")
-                .takes_value(true),
-        )
-        .arg(
-            Arg::with_name(VALIDATORS_ARG_NAME)
-                .long(VALIDATORS_ARG_NAME)
-                .help("Comma separated list of rest endpoints of the validators")
-                .takes_value(true),
-        )
-        .arg(
-            Arg::with_name(WALLET_ADDRESS)
-            .long(WALLET_ADDRESS)
-            .help("The wallet address you will use to bond this mixnode, e.g. nymt1z9egw0knv47nmur0p8vk4rcx59h9gg4zuxrrr9")
-            .takes_value(true)
-            .required(true)
-        )
+#[derive(Args, Clone)]
+pub(crate) struct Init {
+    /// Initialise the mixnode
+    #[clap(long)]
+    id: String,
+
+    /// The host on which the mixnode will be running
+    #[clap(long)]
+    host: String,
+
+    /// The port on which the mixnode will be listening for mix packets
+    #[clap(long)]
+    mix_port: Option<u16>,
+
+    /// The port on which the mixnode will be listening for verloc packets
+    #[clap(long)]
+    verloc_port: Option<u16>,
+
+    /// The port on which the mixnode will be listening for http requests
+    #[clap(long)]
+    http_api_port: Option<u16>,
+
+    /// The custom host that will be reported to the directory server
+    #[clap(long)]
+    announce_host: Option<String>,
+
+    /// Comma separated list of rest endpoints of the validators
+    #[clap(long)]
+    validators: Option<String>,
+
+    /// The wallet address you will use to bond this mixnode, e.g. nymt1z9egw0knv47nmur0p8vk4rcx59h9gg4zuxrrr9
+    #[clap(long)]
+    wallet_address: Option<String>,
+}
+
+impl From<Init> for OverrideConfig {
+    fn from(init_config: Init) -> Self {
+        OverrideConfig {
+            id: init_config.id,
+            host: Some(init_config.host),
+            mix_port: init_config.mix_port,
+            verloc_port: init_config.verloc_port,
+            http_api_port: init_config.http_api_port,
+            announce_host: init_config.announce_host,
+            validators: init_config.validators,
+            wallet_address: init_config.wallet_address,
+        }
+    }
 }
 
-pub async fn execute(matches: ArgMatches<'static>) {
+pub(crate) async fn execute(args: &Init) {
     // TODO: this should probably be made implicit by slapping `#[tokio::main]` on our main method
```

Comment:
When you're doing further refactoring, could you remove that comment? It seems I forgot to get rid of it when I was removing the explicit `tokio::Runtime`

Reply:
Ah yes, good point!